core/src/main/scala/org/apache/spark/executor/Executor.scala

```diff
@@ -432,7 +432,7 @@ private[spark] class Executor(
           setTaskFinishedAndClearInterruptStatus()
           execBackend.statusUpdate(taskId, TaskState.KILLED, ser.serialize(TaskKilled(t.reason)))
 
-        case _: InterruptedException if task.reasonIfKilled.isDefined =>
+        case _: Throwable if task.reasonIfKilled.isDefined =>
```

Comment:
@mridulm, that was my initial reaction as well, but I noticed that we're already catching `Throwable` down on line 447.